src/object/cli_obj.c

```diff
@@ -1669,12 +1669,27 @@ obj_retry_cb(tse_task_t *task, struct dc_object *obj,
 	return rc;
 }
 
+static void
+obj_task_complete(tse_task_t *task, int rc)
+{
+	/* in tse_task_complete only over-write task->dt_result if it is zero, but for some
+	 * cases need to overwrite task->dt_result's retry-able result if get new different
+	 * failure to avoid possible dead loop of retry or assertion.
+	 */
+	if (rc != 0 && task->dt_result != 0 &&
```

Comment:
[minor] "task->dt_result != 0" is redundant.

Reply:
right, can cleanup in following patch, thx